pkg/trace/agent/agent.go

```diff
@@ -381,18 +382,18 @@ func (a *Agent) sample(ts *info.TagStats, pt ProcessedTrace) (events []*pb.Span,
 
 // runSamplers runs all the agent's samplers on pt and returns the sampling decision
 // along with the sampling rate.
-func (a *Agent) runSamplers(pt ProcessedTrace, hasPriority bool) bool {
+func (a *Agent) runSamplers(pt ProcessedTrace, hasPriority, clientDroppedP0s bool) bool {
```

Comment:
Can we put the number of client dropped p0's into the `ProcessedTrace` struct as a field? Bool arguments are a bit of a smell which we've already introduced. Perhaps we can get rid of this one.

We can and probably should do the same for `hasPriority`?